UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaTileMap.cs

```diff
@@ -1165,15 +1167,15 @@ public bool HasOverlay(Vector3Int position, LayerType layerType, OverlayTile ove
 
 			TileLocation tileLocation = null;
 			OverlayTile overlayTile = null;
-			position.z = 1;
+			position.z = 0;
 
 			if (Layers.TryGetValue(layerType, out var layer))
 			{
-				//Only check 20 as its unlikely to ever have more than 20 overlays
+				//Go through overlays under the overlay limit. The first overlay checked will be at z = 1.
```

Comment:
this loop seems to be used a lot and only slightly different each time. Can you extract it into another method to avoid duping code?

Reply:
Not really, because each loop does something different depending on the function it's in

Reply:
yes, just slightly different and as you could experience yourself now when you had to make some changes, you had to do the same changes in all loops. Try extracting the part that is identical in all instances, return what the different methods need and operate with that returned value, that way if we need to change the loop we only change it in one place.